sensors/libsensors-spi.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Sensors SPI communication interface
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <libgen.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <sys/threads.h>
+
+#include <zynq7000-gpio-msg.h>
+
+#include "sensors-spi.h"
+
+
+static struct {
+	handle_t lock; /* SPI bus lock */
+} sensorsspi_common;
+
+
+int sensorsspi_xfer(const spimsg_ctx_t *ctx, oid_t *ss, const void *out, size_t olen, void *in, size_t ilen, size_t iskip)
+{
+	int ret;
+
+	mutexLock(sensorsspi_common.lock);
+
+	/* Manually assert SS line */
+	if (ss != NULL) {
+		gpiomsg_writePin(ss, 0);
+	}
+
+	ret = spimsg_xfer(ctx, out, olen, in, ilen, iskip);
+
+	/* Manually deassert SS line */
+	if (ss != NULL) {
+		gpiomsg_writePin(ss, 1);
+	}
+
+	mutexUnlock(sensorsspi_common.lock);
+
+	return ret;
+}
+
+
+int sensorsspi_initSS(const char *ss)
+{
+	char *path, *dir, *base;
+	unsigned int pin;
+	int err, ntries;
+	oid_t oid;
+
+	path = strdup(ss);
+	if (path == NULL) {
+		return -ENOMEM;
+	}
+
+	/* Get pin number */
+	base = basename(path);
+	if (strncmp(base, "pin", 3)) {
+		free(path);
+		return -EINVAL;
+	}
+	pin = strtoul(base + 3, NULL, 0);
+
+	/* Configure pin as output */
+	dir = dirname(path);
+	sprintf(dir, "%s/dir", dir);
+
+	ntries = 10;
+	while (lookup(dir, NULL, &oid) < 0) {
+		ntries--;
+		if (ntries == 0) {
+			free(path);
+			return -ETIMEDOUT;
+		}
+		usleep(10 * 1000);
+	}
+	err = gpiomsg_writeDir(&oid, 1 << pin, 1 << pin);
```

Comment:
Can't say I like this API - we avoid opening the file, which can cause many problems down the way

Reply:
This one is just a general comment. no need to change it now.